src/components/ScanQRModal/index.tsx

```diff
@@ -28,22 +38,35 @@ const ScanQRModal = ({ classes, isOpen, onClose, onScan }) => {
   useEffect(() => {
     checkWebcam(
       () => {
-        setHasWebcam(true)
+        setUseWebcam(true)
       },
       () => {
-        setHasWebcam(false)
+        setUseWebcam(false)
       },
     )
   }, [])
 
   useEffect(() => {
-    // this fires only when the hasWebcam changes to false (null > false (user doesn't have webcam)
-    // , true > false (user switched from webcam to file upload))
-    // Doesn't fire on re-render
-    if (hasWebcam === false) {
+    if (useWebcam === false && !fileUploadModalOpen && !error) {
+      setFileUploadModalOpen(true)
       openImageDialog()
     }
-  }, [hasWebcam, openImageDialog])
+  }, [useWebcam, openImageDialog, fileUploadModalOpen, setFileUploadModalOpen, error])
```

Comment:
I think this is not needed in dependencies `setFileUploadModalOpen`. But not sure.

Reply:
It's not 100% needed but it's a bad react practice not adding all the things used on the useEffect